presto-docs/src/main/sphinx/optimizer/join-distribution-type.rst

```diff
@@ -0,0 +1,33 @@
+********************************
+Partitioned and Replicated Joins
+********************************
+
+Background
+----------
+Presto can perform two types of distributed joins: partitioned and replicated.
+
+In a partitioned join, both inputs to a join get hash partitioned across the nodes of the cluster. This distribution type is good for larger inputs,
+as it requires less memory on each node. However, it can be much slower than replicated joins because it can require more data to be transferred
```

Comment:
I would say first explain both the types and then come to pros and cons of each type.